### PR TITLE
Update HTML and CSS for new deploy docs

### PR DESCRIPTION
**Pull request by abeatrix**
Closes https://github.com/sourcegraph/sourcegraph/pull/41223 -  Make HTML and CSS on the index page production-quality (Beatrix)

This PR:
- Updates HTML and CSS on the index page production-quality
- Updates HTML and CSS on the AWS One-Click page
- Aligns with other doc components / styles
- Works with both dark and light themes
- Make contents accessible

Light theme:
image

image

Dark theme:
image

image

Next:
- [x] Run through and update deployment docs for each deployment type
- [ ] Add warning note to Kubernetes deployment docs
- [ ] Update link to Sourcegraph Cloud deployment doc when it's ready
- [x] Create Azure deployment docs for Docker Compose? 

## Test plan

To view in local dev,

1. Check out this branch
2. Run yarn docsite:serve from the repo root
3. Point your browser at http://localhost:5080/next

### CODE REVIEW

**Comment by abeatrix** [comment]
Alternatively, we can divided the Cloud provider sections into docker compose and kubernetes section:

image

**Comment by beyang** [comment]
Thanks for cleaning up the HTML and CSS!

A few asks:
- [ ] The cursor over the buttons should change to the "link hand". Probably need this CSS somewhere: `cursor: pointer;`
- [ ] Change the single Docker image button to be the same size as the others—we should discourage this option, since it's non-production.
- [ ] Given that we don't yet have an AWS one-click option, can we change the button to be deactivated / grayed out / with "coming soon" text?
- [ ] Change intro text to "Sourcegraph is runnable in a variety of environments, from Cloud to self-hosted to your local machine.

After these items are addressed, merge is approved.

File: doc/next/index.md [hunk 1]
```diff
@@ -3,194 +3,108 @@ title: Deploy Sourcegraph
...
-	</div>
-	<div class="header">
-		<h1>Docker container</h1>
-	</div>
-	<div class="detail">
-		Spin up Sourcegraph on your local machine.
-	</div>
-</div>
-</a>
+Get started with one of the available deployment methods below, from creating a managed instance hosted by Sourcegraph Cloud to setting up a self-hosted instance with [AWS One-Click](./aws-oneclick.md) --we've got you covered!
```
Comment (beyang) [inline doc/next/index.md:37]:
```suggestion
Sourcegraph is runnable in a variety of environments, from cloud to self-hosted to your local machine.
```

**Comment by abeatrix** [comment]
@beyang As discussed, the new deployment page is now the default homepage for docs site
image

AWS One Click installation launcher is currently disabled with a `coming soon` label added:
image

File: doc/index.md [hunk 2]
```diff
@@ -1,50 +1,152 @@
...
-<p class="lead">
-Understand, fix, and automate across your codebase with Sourcegraph.
-</p>
+# Get Started
 
-### Try Sourcegraph
+## Deploy Sourcegraph
 
-You can quickly try Sourcegraph locally using Docker:
+Sourcegraph is runnable in a variety of environments, from cloud to self-hosted to your local machine.
```
Comment (beyang) [inline doc/index.md:41]:
```suggestion
Sourcegraph is runnable in a variety of environments, from cloud to self-hosted to your local machine.

* We recommend Sourcegraph Cloud for all new customers.
* If you prefer to self-host, we recommend following one of our single-node setup docs on AWS, GCP, Azure, Digital Ocean, or a generic Linux VM.
* For larger enterprises that require multi-node deployments, we offer a set of Kubernetes manifests. We strongly recommend getting in touch with us in our [Discord]() or [over email](support@sourcegraph.com) if you choose to pursue this option.
```

**Review by beyang — APPROVED** [review]

**Pull request merged**